jaxrs/src/main/java/com/cerner/beadledom/jaxrs/provider/ForwardedHeaderFilter.java

```diff
@@ -0,0 +1,72 @@
+package com.cerner.beadledom.jaxrs.provider;
+
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Priority;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.PreMatching;
+import javax.ws.rs.ext.Provider;
+
+/**
+ * The ForwardedHeaderFilter reads the request context and determines if the client request was made
+ * from a secure protocol (HTTPS). If the request originated from secure context, it updates the
+ * request to reflect the original protocol.
+ *
+ * <p>To determine if the client request made in a secure context was forwarded by a load balancer
+ * or proxy, the "Forwarded" and "X-Forwarded-Proto" values are extracted and checked.</p>
+ *
+ * @author Nick Behrens
+ */
+@Provider
+@Priority(Priorities.HEADER_DECORATOR)
+@PreMatching
+public class ForwardedHeaderFilter implements ContainerRequestFilter {
+
+  @Override
+  public void filter(ContainerRequestContext requestContext) throws IOException {
+    String httpScheme = requestIsSecure(requestContext) ? "https" : "http";
+    requestContext.setRequestUri(
+        requestContext.getUriInfo().getRequestUriBuilder().scheme(httpScheme).build());
+  }
+
+  /**
+   * Checks if the request was made from a secure context or not.
+   * @param requestContext The {@link ContainerRequestContext} object containing the container request headers.
+   * @return true if the request was made from a secure context, false otherwise.
+   */
+  private boolean requestIsSecure(ContainerRequestContext requestContext) {
+    return requestContext.getSecurityContext().isSecure()
+        || hasSecureForwardedHeader(requestContext)
+        || hasSecureXForwardedProtoHeader(requestContext);
+  }
+
+  /**
+   * Checks the Forwarded header for a protocol value of https.
+   * @param requestContext The {@link ContainerRequestContext} object containing the container request headers.
+   * @return true if the `Forwarded` header is present and contains https; false otherwise.
+   */
+  private boolean hasSecureForwardedHeader(ContainerRequestContext requestContext) {
+    String forwardedHeader = requestContext.getHeaderString("Forwarded");
+    if (forwardedHeader == null) {
+      return false;
+    }
+    Pattern forwardedPairs = Pattern.compile("proto=(?<protocolValue>[^;]*)(;|\\z)");
```

Comment:
could we add a doc explaining this regex because its not clear

Reply:
Should it also be extracted out to the class level so the pattern isn't recreated on every request?

Reply:
Yep, it should be. I moved it out and added a little documentation on what it does here: https://github.com/cerner/beadledom/pull/149/commits/9fd05e06fd0a408492e73aabea311542f32f36ac